extensions/notebook/src/test/model/sessionManager.test.ts

```diff
@@ -166,4 +181,126 @@ describe('Jupyter Session', function (): void {
 		should(kernel.id).equal('id');
 		should(options.name).equal('python');
 	});
+
+	it('should write configuration to config.json file', async function (): Promise<void> {
+		let tempDir = path.join(os.tmpdir(), '.sparkmagic');
+		let configPath = path.join(tempDir, 'config.json');
+		const expectedResult = {
+			'kernel_python_credentials': {
+				'url': 'http://localhost:8088'
+			},
+			'kernel_scala_credentials': {
+				'url': 'http://localhost:8088'
+			},
+			'kernel_r_credentials': {
+				'url': 'http://localhost:8088'
+			},
+			'livy_session_startup_timeout_seconds': 100,
+			'logging_config': {
+				'version': 1,
+				'formatters': {
+					'magicsFormatter': {
+						'format': '%(asctime)s\t%(levelname)s\t%(message)s',
+						'datefmt': ''
+					}
+				},
+				'handlers': {
+					'magicsHandler': {
+						'class': 'hdijupyterutils.filehandler.MagicsFileHandler',
+						'formatter': 'magicsFormatter',
+						'home_path': ''
+					}
+				},
+				'loggers': {
+					'magicsLogger': {
+						'handlers': ['magicsHandler'],
+						'level': 'DEBUG',
+						'propagate': 0
+					}
+				}
+			},
+			'ignore_ssl_errors': true,
+		};
+		expectedResult.logging_config.handlers.magicsHandler.home_path = tempDir;
+		sinon.stub(utils, 'getUserHome').returns(tempDir);
+		await session.configureKernel();
+		let result = await fs.promises.readFile(configPath, 'utf-8');
+		should(JSON.parse(result) === expectedResult);
+		sinon.restore();
```

Comment:
Do the sinon.restore in an afterEach hook

https://github.com/microsoft/azuredatastudio/blob/b2a1738836d250e54814fae556b9160674e7f87c/extensions/arc/src/test/ui/dialogs/connectControllerDialog.test.ts#L14